flowmachine/flowmachine/core/server/query_proxy.py

```diff
@@ -510,11 +504,32 @@ def get_sql(self):
         """
         query_id = self._get_query_id_from_redis()
         try:
-            if self.redis_interface.has_lock(query_id):
-                raise QueryProxyError(f"Query with id '{query_id}' is still running.")
-            else:
-                return get_sql_for_query_id(query_id)
-        except (AttributeError, ValueError):
+            query_state = self.redis_interface.query_status(query_id)
+        except ValueError as e:
+            raise QueryProxyError(
+                f"Got a bad state for '{query_id}'. Original exception was {e}"
+            )
+
+        if query_state == QueryState.EXECUTING:
```

Comment:
Just as a thought, I wonder whether it would be useful to have the error messages live in QueryState directly and simply propagate them here? Would allow us to get rid of the if/elif chain. But haven't looked into it deeply, it's possble this wouldn't work out.